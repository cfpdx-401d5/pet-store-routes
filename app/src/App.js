import React, { Component } from 'react';
import Navigation from './components/Navigation';
import Home from './components/Home';

class App extends Component {
    render() {
        return (
          <div>
            <Navigation />
            <Home />
          </div>
        );
    }
}

export default App;
