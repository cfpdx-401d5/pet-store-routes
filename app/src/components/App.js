import React from 'react';
import { Link } from 'react-router-dom';
import Navigation from './Navigation';

class App extends React.Component {
    render() {
        return (
            <div>
                <Navigation />
                <h1>Welcome to Pets R Kewl</h1>
                <Link to='/stores'>View Store Locations></Link>
            </div>
        );
    }
}

export default App;
