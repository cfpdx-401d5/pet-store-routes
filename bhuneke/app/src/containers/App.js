import React from 'react';
import NavBar from '../components/NavBar';
import PageBody from '../components/PageBody';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import fetcher from '../helpers/fetcher';

export default class App extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      stores: [],
    };
  }

  doFetch() {
    fetcher({
      path: '/stores',
      method: 'GET',
    })
    .then(res => {
      return res.json();
    })
    .then(stores => {
      this.setState({ stores });
    });
  }

  componentDidMount() {
    this.doFetch();
  }

  render() {
    return (
      <Router>
        <div>
          < Route path='/' component={ NavBar } />
          < PageBody stores={ this.state.stores } />
        </div>
      </Router>
    );
  }
}
