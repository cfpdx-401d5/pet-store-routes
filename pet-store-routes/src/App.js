import React from 'react'
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom'
import Home from './pages/Home';
import Store from './pages/Store';
import StoreOne from './pages/StoreOne';
import StoreTwo from './pages/StoreTwo';


export default function App () {
  return (
  <Router>
    <div>
      <ul>
        <li><Link to="/">Home</Link></li>
        <li><Link to="/stores">Store</Link></li>
      </ul>

      <hr/>

      <Route exact path="/" component={Home}/>
      <Route path="/stores" component={Store}/>
      <Route path="/stores/store_one" component={StoreOne}/>
      <Route path="/stores/store_two" component={StoreTwo}/>
    </div>
  </Router>
  )
}
