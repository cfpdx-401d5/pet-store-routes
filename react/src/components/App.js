import React, { Component } from 'react';
import Selector from './Selector';
import { Route } from 'react-router-dom';
import Home from './Home';
import Signup from './Signup';
import Signin from './Signin';
import Stores from './Stores';
import Store from './Store';



export default class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            pageView: 'home',
            viewStyle: null,
        };
    }


    render() {
        return (
            <div className="App">
              <Selector/>
              <Route exact path="/" component={Home}/>
              <Route exact strict path="/signup" component={Signup}/>
              <Route exact strict path="/signin" component={Signin}/>
              <Route exact strict path="/stores" component={Stores}/>
              <Route path="/stores/:id" component={Store}/>
            </div>
        );
    }
}
