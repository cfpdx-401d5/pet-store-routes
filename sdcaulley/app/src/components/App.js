import React from 'react';
import { Link } from 'react-router-dom';

class App extends React.Component {
    render() {
        return (
            <div>
                <h1>Welcome to Pets R Kewl</h1>
                <Link to='/stores'>View Store Locations></Link>
            </div>
        );
    }
}

export default App;
