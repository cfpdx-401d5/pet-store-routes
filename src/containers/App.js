import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import Home from '../components/Home';
import Stores from '../components/Stores';
import Store from '../components/Store';




class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      stores: [{
        store_id: 1,
        location: 'downtown',
      },
      {
        store_id: 2,
        location: 'midtown',
      },
      {
        store_id: 3,
        location: 'uptown',
      }],
      pets: [
        {
          pet_id: 11,
          name: 'george',
          storeId: 1,
          type: 'cat',
          cost: '$2'
        },
        {
          pet_id: 12,
          storeId: 1,
          name: 'petunia',
          type: 'parakeet',
          cost: '$12'
        },
        {
          pet_id: 24,
          storeId: 2,
          name: 'hairy',
          type: 'sloth',
          cost: '$3'
        }, {
          pet_id: 21,
          storeId: 2,
          name: 'gladys',
          type: 'cat',
          cost: '$7'
        },
        {
          pet_id: 22,
          storeId: 2,
          name: 'slipper',
          type: 'parakeet',
          cost: '$17'
        },
        {
          pet_id: 34,
          storeId: 3,
          name: 'forager',
          type: 'sloth',
          cost: '$6'
        },
      ]
    }
  }
  render() {
    return (
      <div>
        <Route exact path='/' component={Home} />
        <Route path='/stores' render={(props) => (
          <Stores { ...props }
            stores={this.state.stores}
            pets={this.state.pets} />
        )} />

        <Route path='/store/:store_id' render={(props) => (
          <Store { ...props }
            stores={this.state.stores}
            pets={this.state.pets} />
        )} />
      </div>
    );
  }
}

export default App;
