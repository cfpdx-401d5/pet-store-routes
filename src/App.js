import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom';

import Home from './components/Home';
import { Stores } from './components/Stores';

import './App.scss';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      petStores: [{
        name: 
        location:
      }, {

      }, {

      }]
    };
  }

  render() {
    return (
      <Router>
        <nav>
          <ul>
            <li><Link to='/'>Home</Link></li>
            <li><Link to='/stores'>Stores</Link></li>
          </ul>

          <hr/>
          
          <Route exact path='/' component={Home}/>
          <Route path='/stores' component={Stores}/>
        </nav>

      </Router>
    );
  }
}


export default App;
