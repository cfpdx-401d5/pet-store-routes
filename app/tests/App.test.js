/* eslint-disable */
import React from 'react';
import {shallow, render} from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../src/App';

describe('Full App', () => {
    it('full app display', () => {
        const rendered = shallow(<App />);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});