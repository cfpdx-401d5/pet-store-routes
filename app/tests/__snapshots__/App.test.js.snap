exports[`Full App full app display 1`] = `
<div>
  <Navigation />
</div>
`;
